### PR TITLE
ASTT-72 version the astt software

### PR DESCRIPTION
**Pull request by AbednigoLethole**

### CODE REVIEW

File: .github/workflows/asst-ci.yml [hunk 1]
```diff
@@ -13,136 +16,143 @@ jobs:
     steps:
       - name: Checkout Repository
         uses: actions/checkout@v2
+    
+      - name: Test env GITHUB_REF_NAME
+        run: echo "GITHUB_REF_NAME=$GITHUB_REF_NAME"
```
Comment (Silere8) [inline .github/workflows/asst-ci.yml:21]: If we use github.ref_name as the environment variable for checking the branch name, is line 20 necessary?

Comment (AbednigoLethole) [reply .github/workflows/asst-ci.yml:21]: No it's not necessary. I will remove it.

It was for testing purposes.

Comment (Silere8) [reply .github/workflows/asst-ci.yml]: Awesome.

File: .github/workflows/asst-ci.yml [hunk 2]
```diff
@@ -14,135 +17,164 @@ jobs:
...
-      uses: actions/setup-python@v2
-      with:
-        python-version: '3.10'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install poetry
+          poetry config virtualenvs.create false && poetry install
 
-    - name: Install dependencies
```
Comment (Silere8) [inline .github/workflows/asst-ci.yml:40]: I see we do not have a formal lint job, I think I have to configure it and group all the lint tools we will use. That being said, for mypy will it be a job as a stand alone?

File: .github/workflows/asst-ci.yml [hunk 3]
```diff
@@ -14,135 +17,164 @@ jobs:
...
-
-    - name: Login to GitHub Container Registry
-      run: echo "${{ secrets.ASTT_PASS }}" | docker login ghcr.io --username "${{ secrets.ASTT_USR }}" --password-stdin
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.10'
```
[the remark this answers is not in the document]
Comment (Silere8) [reply .github/workflows/asst-ci.yml:82]: [See](https://github.com/actions/starter-workflows/blob/main/ci/pylint.yml) and [this](https://stackoverflow.com/questions/65945929/poetry-how-to-publish-project-packages-targeting-multiple-python-versions)

[the remark this answers is not in the document]
Comment (Silere8) [reply .github/workflows/asst-ci.yml:82]: We can test it out to see if it does not break our workflow.

File: .github/workflows/asst-ci.yml (hunk 2 above)
Comment (AbednigoLethole) [reply .github/workflows/asst-ci.yml:40]: @Silere8  python-lint is the formal lint job. We can add mypy to the job.

File: .github/workflows/asst-ci.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (AbednigoLethole) [reply .github/workflows/asst-ci.yml:82]: Please try out the changes and push

**Comment by USilere** [comment]
LGTM.

**Review by PhelokaziDube — APPROVED** [review]

**Pull request merged**